### PR TITLE
Add a ranking system when an event state change to ENDED

### PR DESCRIPTION
**Pull request by vdumestre**
Implementation of #23

### CODE REVIEW

File: src/main/java/tv/zerator/ffs/api/v1/resources/EventResource.java [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package tv.zerator.ffs.api.v1.resources;
 
 import java.sql.SQLException;
+import java.util.*;
```
Comment (Hartorn) [inline src/main/java/tv/zerator/ffs/api/v1/resources/EventResource.java]: Not recommended to use wildcard import in Java, most of the formatter / codestyle rule prefer unit import

File: src/main/java/tv/zerator/ffs/api/v1/resources/EventResource.java [hunk 2]
```diff
@@ -69,12 +78,80 @@ public Status deleteEvent() throws SQLException {
...
+		}
+
+		for (Integer userId:
+			 lastUsers) {
+			mEvents.updateUserRank(mEventId, userId, i);
+		}
+
+	}
+
+	class ScoreUserValueComparator implements Comparator<Map.Entry<Integer, Double>> {
```
Comment (Hartorn) [inline src/main/java/tv/zerator/ffs/api/v1/resources/EventResource.java]: Can't this class be private static ?

File: src/main/java/tv/zerator/ffs/api/v1/resources/EventResource.java [hunk 3]
```diff
@@ -69,12 +78,80 @@ public Status deleteEvent() throws SQLException {
...
+		List<Map.Entry<Integer,Double>> list = new ArrayList<>(map.entrySet());
+		Collections.sort(list, new ScoreUserValueComparator(event.getRankingType()));
+
+		int i = 1;
+		int currentRank = 1;
+		Double currentScore = null ;
+		for (Map.Entry<Integer,Double> entry:
+				list) {
+
+			if (currentScore == null || !currentScore.equals(entry.getValue())){
```
Comment (Hartorn) [inline src/main/java/tv/zerator/ffs/api/v1/resources/EventResource.java]: Can't this be simplified as `if (!entry.getValue().equals(currentScore))`

File: src/main/java/tv/zerator/ffs/api/v1/resources/EventResource.java [hunk 4]
```diff
@@ -69,12 +78,80 @@ public Status deleteEvent() throws SQLException {
...
+
+		for (Integer userId:
+			 lastUsers) {
+			mEvents.updateUserRank(mEventId, userId, i);
+		}
+
+	}
+
+	class ScoreUserValueComparator implements Comparator<Map.Entry<Integer, Double>> {
+		EventBean.RankingType rankingType;
```
Comment (Hartorn) [inline src/main/java/tv/zerator/ffs/api/v1/resources/EventResource.java]: This should be private final

Comment (Hartorn) [reply src/main/java/tv/zerator/ffs/api/v1/resources/EventResource.java]: Also, when this is private final, you can remove constructor and use lombok `@RequiredArgsConstructor`

File: src/main/java/tv/zerator/ffs/api/v1/resources/EventResource.java (hunk 1 above)
Comment (vdumestre) [reply src/main/java/tv/zerator/ffs/api/v1/resources/EventResource.java]: changed

File: src/main/java/tv/zerator/ffs/api/v1/resources/EventResource.java (hunk 3 above)
Comment (vdumestre) [reply src/main/java/tv/zerator/ffs/api/v1/resources/EventResource.java]: i think this would return a NullPointerException for the first user

**Comment by Hartorn** [comment]
Looks good to me ! :+1:

**Comment by AlexMog** [comment]
Seems ok on my side ! Merging !
Thanks for your help :)
Closes #23

**Pull request merged**